Контроль и ограничения (часть 2)

S-E-Pashin opened the pull request:
…олучаемыми оферами, сортирует в зависимости от выбранного положения. Взаимодействует с хранилищем.

Реализует поведение наведения на карточку и отображение активного маркера данного предложения на карте. Внедряет функциональность отображения подготовленного компонента в случае отсутствия предложений для данного(выбранного) города. (Так же интерактивно)

---
:mortar_board: [Контроль и ограничения (часть 2)](https://up.htmlacademy.ru/react-individual/2/user/1072207/tasks/9)

S-E-Pashin committed:
Создает компонент с функциональностью сортировки. Взаимодействует с получаемыми оферами, сортирует в зависимости от выбранного положения. Взаимодействует с хранилищем.
Реализует поведение наведения на карточку и отображение активного маркера данного предложения на карте.
Внедряет функциональность отображения подготовленного компонента в случае отсутствия предложений для данного(выбранного) города. (Так же интерактивно)

Finnko commented on src/components/container-not-offers/container-not-offers.tsx:
```diff
@@ -0,0 +1,21 @@
+import {CitiesCoordinatesKeys} from '../../const/city-points.ts';
+import {useAppSelector} from '../../hooks/use-app-selector.ts';
+
+function ContainerNotOffers() {
```
Я бы использовал лучше без отрицания Empty и аналогичные

Finnko commented on src/components/container-not-offers/container-not-offers.tsx:
```diff
@@ -0,0 +1,21 @@
+      <section className="cities__no-places">
+        <div className="cities__status-wrapper tabs__content">
+          <b className="cities__status">No places to stay available</b>
+          <p className="cities__status-description">We could not find any property available at the moment in {` ${selectedCityStore}`}
```
Нам не нужно оборачивать в бектики и делать интерполяцию
```suggestion
          <p className="cities__status-description">We could not find any property available at the moment in {selectedCityStore}
```

Finnko commented on src/components/offers-list/offers-list.tsx:
```diff
@@ -28,8 +30,14 @@ function OffersList({offers, listType}: OffersListProps): JSX.Element {
-            onMouseToCard={() => setCard(item.id)}
-            onMouseLeaveCard={() => setCard(null)}
+            onMouseToCard={() => {
+              setCard(item.id);
+              store.dispatch(updateActiveOffer(item.id));
```
Надо определиться или в стейте компонента хранить или в сторе, зачем и там и там

Finnko commented on src/components/offers-list/offers-list.tsx:
```diff
@@ -28,8 +30,14 @@ function OffersList({offers, listType}: OffersListProps): JSX.Element {
+            }}
+            onMouseLeaveCard={() => {
+              setCard(null);
+              store.dispatch(updateActiveOffer(''));
```
Store не надо импортировать в каждый компонент чтобы получить диспатч у тебя есть хук useAppDispatch()

S-E-Pashin committed:
Изменяет подход к отображению и передаче данных в/из хранилища.
Сохраняет исходные данные в хранилище и работает с получаемыми данными.
Фильтрует, сортирует и передает в компонент/ы данные для отображения.
Изменяет логику компонента фильтр. Мапит содержимое перечня сортировки и функций взаимодействия с ними. Использует диспатч - юз. диспатч.
Внедряет хук закрывающий перечень сортировки при клике вне его области.

S-E-Pashin committed:
Рефакторинг кастомного хука, типов, нейминга.
